apis/externalsecrets/v1/provider_schema_maintenance.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"sync"
+)
+
+type MaintenanceStatus bool
+
+const (
+	MaintenanceStatusMaintained    MaintenanceStatus = true
+	MaintenanceStatusNotMaintained MaintenanceStatus = false
+)
+
+var maintenance map[string]MaintenanceStatus
```

Comment:
Same Logic for provider Registration, but for maintenance status.

This is used by `Register`/`ForceRegister` methods now.